src/bank01_player.asm

```diff
@@ -697,18 +693,70 @@ scrollRoom:
     set  2, A                                          ;; 01:4502 $cb $d7
     ld   [wMainGameStateFlags.nextFrame], A            ;; 01:4504 $ea $a2 $c0
     ld   [wMainGameStateFlags], A                      ;; 01:4507 $ea $a1 $c0
-.jr_01_450a:
-    ld   A, E                                          ;; 01:450a $7b
-    push DE                                            ;; 01:450b $d5
-    bit  0, A                                          ;; 01:450c $cb $47
-    jr   NZ, .west                                     ;; 01:450e $20 $31
-    bit  1, A                                          ;; 01:4510 $cb $4f
-    jp   NZ, .east                                     ;; 01:4512 $c2 $97 $45
-    bit  2, A                                          ;; 01:4515 $cb $57
-    jp   NZ, .south                                    ;; 01:4517 $c2 $e7 $45
-    bit  3, A                                          ;; 01:451a $cb $5f
-    jp   NZ, .north                                    ;; 01:451c $c2 $3a $46
-    pop  DE                                            ;; 01:451f $d1
+    ld   A, D
+    ld   [wSpriteScrollSpeed], A
+
+    ; Set B to be the flip of the direction in E
+    ld   B, DIRECTIONF_EAST
+    bit  DIRECTIONB_WEST, E
+    jr   NZ, .check_room_override
+    sla  B ; B holds DIRECTIONF_WEST
+    bit  DIRECTIONB_EAST, E
+    jr   NZ, .check_room_override
+    sla  B ; B holds DIRECTIONF_NORTH
+    bit  DIRECTIONB_SOUTH, E
+    jr   NZ, .check_room_override
+    sla  B ; B holds DIRECTIONF_SOUTH
+.check_room_override:
+    ld   HL, wNextRoomOverride
+    push DE
+    ld   A, [HL+]
+    ld   E, A
+    ld   D, [HL]
+
+    ; If a room override is not set, DE=$ffff
+    and  A, D
+    inc  A
+
+    ld   A, B ; sets which direction to move
+    jr   Z, .choose_next_room
+    or   A, $80 ; sets flag to use override
+.choose_next_room:
+    or   A, $10 ; sets flag to choose next room
+    call call_00_2617
+    pop  DE
+    ld   A, $ff
+    ld   HL, wNextRoomOverride
+    ld   [HL+], A
+    ld   [HL], A
+    ld   C, A ; indicates this is the first pass through scrollRoom function for this transition
+.scroll_started:
+    ld   A, E
+    and  A, $0f
+    jr   Z, .stop_scroll
+
+    ; On the first pass we know we will not be moving the screen, so ignore other checks
+    and  A, C
+    jr   NZ, .scrollRoomMove
+
+    ; If VBlankDone is over 1, then let it catch up before continuing to scroll
+    ld   A, [wVBlankDone]
+    cp   A, $02
+    ret  NC
+
+    ; If we've started scrolling, ignore the wTileCopyRequestCount check
+    ld   A, [wScrollPixelCounter]
+    and  A, A
+    jr   NZ, .scrollRoomMove
+
+    ; Verify that all tiles on new map are written before beginning scroll
```

Comment:
I assume this check should be removed?

Reply:
I don't think so? It only does this check when wScrollPixelCounter is 0. I think it's important for the tile data to be in place before we begin the scroll operation.